### PR TITLE
SMS Test Improvements

### PR DESCRIPTION
**Pull request by gcapalbo**
The main change here is using setUpClass/tearDownClass instead of setUp/tearDown. Whe run locally, takes 10s off opt_tests and 52s off test_backends.

@emord @benrudolph

### CODE REVIEW

**Review by benrudolph — APPROVED** [review]
💬 🏃  looks good!

File: corehq/apps/sms/tests/test_backends.py [hunk 1]
```diff
@@ -401,181 +421,152 @@ def test_push_inbound_sms(self):
...
-        self.grapevine_backend.delete()
-        self.twilio_backend.delete()
-        self.megamobile_backend.delete()
-        self.smsgh_backend.delete()
-        self.apposit_backend.delete()
-        self.sislog_backend.delete()
-        self.yo_backend.delete()
-        self.push_backend.delete()
-        self.icds_backend.delete()
-        super(AllBackendTest, self).tearDown()
```
Comment (benrudolph) [inline corehq/apps/sms/tests/test_backends.py]: 👏

File: corehq/apps/sms/tests/test_backends.py [hunk 2]
```diff
@@ -37,127 +39,145 @@
...
 
-    def setUp(self):
-        super(AllBackendTest, self).setUp()
-
-        self.domain_obj = Domain(name='all-backend-test')
-        self.domain_obj.save()
-        self.create_account_and_subscription(self.domain_obj.name)
-        self.domain_obj = Domain.get(self.domain_obj.get_id)
+    @classmethod
+    def setUpClass(cls):
```
Comment (emord) [inline corehq/apps/sms/tests/test_backends.py]: needs to call super's setupclass. same for teardownclass. there's a warning right after this box: https://docs.djangoproject.com/en/1.10/topics/testing/tools/#django.test.SimpleTestCase.allow_database_queries

Comment (gcapalbo) [inline corehq/apps/sms/tests/test_backends.py]: ah didn't realize it was required. fixed here: https://github.com/dimagi/commcare-hq/pull/15149/commits/37ca44397450734dab3b19d091ebcabbc98d0fee

**Pull request merged**